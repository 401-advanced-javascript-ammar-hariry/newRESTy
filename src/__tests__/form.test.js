import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Footer from '../components/footer/index';
import Header from '../components/header/index';
import Form from '../components/form/form';


describe('<footer />', () => {
    it('<footer /> is exists', () => {

        let app = shallow( <Footer /> );
        expect(app.find('footer').exists()).toBeTruthy
        expect(app.find('footer').text()).toContain('Fellows')
         });
});
describe('<header />', () => {
	it('<header /> is exists', () => {
      
	    let app = shallow( <Header /> );
	    expect(app.find('header').exists()).toBeTruthy
	    expect(app.find('header').text()).toContain('RESTy')
	});
 });
 describe('<Form />', () => {
	it('<Form /> is exists', () => {
      
	    let app = shallow( <Form /> );
	    expect(app.find('form').exists()).toBeTruthy

	});
	it(' it stores the users input into state', () => {
		let form = shallow( <Form /> );
		const input1 =form.find('input');
		input1.simulate('change', {target: {value: 'testing'}});
		expect(form.state('url')).toBe('testing');
	});
	it(' it clear the form/state after the form is submitted', () => {
		let form = shallow( <Form /> );
		const button =form.find('button');
		button.simulate('click');
		expect(form.state('url')).toBe('');
	});
	it(' it  renders the correct component', () => {
		const tree = renderer.create(<Form />).toJSON();
		expect(tree).toMatchSnapshot();
	});

      
 });