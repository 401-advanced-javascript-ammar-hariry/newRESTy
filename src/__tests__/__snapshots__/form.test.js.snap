// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form />  it  renders the correct component 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <label>
      <span>
        URL: 
      </span>
      <input
        name="url"
        onChange={[Function]}
        type="text"
      />
      <button
        type="submit"
      >
        GO!
      </button>
    </label>
    <label
      className="methods"
    >
      <span
        className=""
        id="get"
        onClick={[Function]}
      >
        GET
      </span>
      <span
        className=""
        id="post"
        onClick={[Function]}
      >
        POST
      </span>
      <span
        className=""
        id="put"
        onClick={[Function]}
      >
        PUT
      </span>
      <span
        className=""
        id="delete"
        onClick={[Function]}
      >
        DELETE
      </span>
    </label>
  </form>,
  <section
    className="results"
  >
    <span
      className="method"
    />
    <span
      className="url"
    />
  </section>,
]
`;

exports[`<Form />  it clear the form/state after the form is submitted 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <label>
      <span>
        URL: 
      </span>
      <input
        name="url"
        onChange={[Function]}
        type="text"
      />
      <button
        type="submit"
      >
        GO!
      </button>
    </label>
    <label
      className="methods"
    >
      <span
        className=""
        id="get"
        onClick={[Function]}
      >
        GET
      </span>
      <span
        className=""
        id="post"
        onClick={[Function]}
      >
        POST
      </span>
      <span
        className=""
        id="put"
        onClick={[Function]}
      >
        PUT
      </span>
      <span
        className=""
        id="delete"
        onClick={[Function]}
      >
        DELETE
      </span>
    </label>
  </form>,
  <section
    className="results"
  >
    <span
      className="method"
    />
    <span
      className="url"
    />
  </section>,
]
`;
